src/runtime_src/core/pcie/tools/xbutil/xbutil.h

```diff
@@ -369,6 +370,30 @@ class device {
         return 0;
     }
 
+    int getPSKernels(std::vector<ps_kernel_data> &psKernels) const
+    {
+        std::string errmsg;
+        std::vector<char> buf;
+
+        pcidev::get_dev(m_idx)->sysfs_get("icap", "ps_kernel", errmsg, buf);
+
+        if (!errmsg.empty()) {
+            std::cout << errmsg << std::endl;
+            return -EINVAL;
+        }
+        if (buf.empty())
```

Comment:
Should this return 0 or -EINVAL?

Reply:
It should be 0. No content in ps_kernel sysfs node